bemserver_core/model/sites.py

```diff
@@ -393,7 +416,23 @@ def register_class(cls):
             },
         )
 
-
+    def _before_flush(self):
+        # Get property type and try to cast value to ensure its validity.
+        site_prop = SiteProperty.get_by_id(self.site_property_id)
+        if site_prop is not None:
+            try:
+                self.value = site_prop.structural_element_property.value_type.cast(
+                    self.value
+                )
+            except ValueError as exc:
+                self.value = None
+                prop_name = site_prop.structural_element_property.name
+                raise PropertyTypeInvalidError(
```

Comment:
I don't think we actually use the exception message anywhere in the code. If we don't mind a more generic message (without `prop_name`) we could try/except/raise in the enum cast method to keep the code here short.

Reply:
OK, move the try/except/raise in enum cast method and raise a more generic message.
Don't need to put `prop_name`in message, in principle the call to the cast method is already made in the context of the use of a property for which we already have all the information (including its name).